apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -222,12 +222,18 @@ namespace MWRender
         resourceSystem->getSceneManager()->setForceShaders(Settings::Manager::getBool("force shaders", "Shaders"));
         resourceSystem->getSceneManager()->setClampLighting(Settings::Manager::getBool("clamp lighting", "Shaders"));
         resourceSystem->getSceneManager()->setForcePerPixelLighting(Settings::Manager::getBool("force per pixel lighting", "Shaders"));
+        resourceSystem->getSceneManager()->setAccurateFog(Settings::Manager::getBool("accurate fog", "Shaders"));
         resourceSystem->getSceneManager()->setAutoUseNormalMaps(Settings::Manager::getBool("auto use object normal maps", "Shaders"));
         resourceSystem->getSceneManager()->setNormalMapPattern(Settings::Manager::getString("normal map pattern", "Shaders"));
         resourceSystem->getSceneManager()->setNormalHeightMapPattern(Settings::Manager::getString("normal height map pattern", "Shaders"));
         resourceSystem->getSceneManager()->setAutoUseSpecularMaps(Settings::Manager::getBool("auto use object specular maps", "Shaders"));
         resourceSystem->getSceneManager()->setSpecularMapPattern(Settings::Manager::getString("specular map pattern", "Shaders"));
 
+        if (resourceSystem->getSceneManager()->getClampLighting())
```

Comment:
Won't this cause shaders to be used when the lighting *is* clamped?

Reply:
While we're discussing this, we only globablly applied the setting for terrain. The documentation for the setting says
```ini
# Restrict the amount of lighting that an object can receive to a maximum of (1,1,1).
# Only affects objects that render with shaders (see 'force shaders' option). Always affects terrain.
# Setting this option to 'true' results in fixed-function compatible lighting, but the lighting
# may appear 'dull' and there might be color shifts.
# Setting this option to 'false' results in more realistic lighting.
clamp lighting = true
```

The reason it's applied to *terrain* specifically is because if one terrain chunk uses the shader as it's got a normal map, we need to ensure there're no lighting seams with adjacent terrain chunks. This isn't something that can happen with objects as they're distinct things, not multiple things pretending to be one, so I'm not 100% sure if it's a good idea to apply everywhere.

If we *do* decide this approach is better, we need to make sure the documentation is updated.

Reply:
If a person has hardware capable of GLSL, why would they run it selectively on terrain only? openmw is bottlenecked on the CPU anyway, and the fixed pipeline can be slower since no one cares about its implementation anymore.

I recommend centralizing GPU-related options changing other options. That makes sure they're always at consistent values, in addition to being more readable overall.

Reply:
I *think* we still technically support hardware that *supports* GLSL, but incredibly slowly. Even on modern hardware, fixed-function isn't always slower than the mish-mash of deprecated fixed-function and shader-based stuff we do - when vendors optimise their shader-based paths, I doubt they put much effort into making sure the optimisations work when deprecated behaviour is in use, but that deprecated behaviour was mainstream when they last touched the fixed-function stuff, so provided they're not doing anything to make it worse, we can be better off doing an old and deprecated normal thing than a half-and-half weird thing.